Applications/WebPortal/WebApp/library/source/pages/account_page/entitlements_page/entitlements_page.tsx

```diff
@@ -13,28 +16,95 @@ interface Properties {
   /** The set of entitlements that are checked. */
   checked: Beam.Set<Beam.DirectoryEntry>;
 
+  /** The database of currencies */
+  currencyDatabase: Nexus.CurrencyDatabase;
+
+  /** The set of markets. */
+  marketDatabase: Nexus.MarketDatabase;
+
+  /** The size at which the component should be displayed at. */
+  displaySize: DisplaySize;
+
+  /** The message that lets the user know if their submission was succesful. */
+  submissionStatus?: string;
+
   /** Indicates an entitlement has been clicked.
    * @param entitlement - The entitlement that was clicked.
    */
-  onEntitlementClick?: (entitlement: Beam.DirectoryEntry) => void;
+  onEntitlementClick?(entitlement: Beam.DirectoryEntry): void;
 
   /** Indicates the form should be submitted.
    * @param comment - The comment to submit with the form.
    */
-  onSubmit?: (comment: string) => void;
+  onSubmit?(comment: string): void;
 }
 
+/* Displays a list of entitlements. */
 export class EntitlementsPage extends React.Component<Properties> {
   public static readonly defaultProps = {
     onEntitlementClick: () => {},
     onSubmit: () => {}
-  }
+  };
 
   constructor(props: Properties) {
     super(props);
   }
 
   public render(): JSX.Element {
-    return <div/>;
+    const entitlements = (() => {
+      const rows = [];
+      for(const entry of this.props.entitlements) {
+        rows.push(
+          <EntitlementRow
+            entitlementEntry={entry}
+            currencyEntry=
+            {this.props.currencyDatabase.fromCurrency(entry.currency)}
+            isActive={this.props.checked.test(entry.group)}
+            onClick = {() => {
+              this.props.onEntitlementClick(entry.group);
+              this.forceUpdate();
```

Comment:
This ```forceUpdate``` should not be needed, we have to find a way to remove it. The component that installed the ```onEntitlementClick``` callback should be responsible for updating this component.

Reply:
The problem has to do with arrays and objects as props. When I change the values/states inside of an object or an array the changes don't seem to trickle down like they do for other props.